src/pybind/mgr/rbd_support/mirror_snapshot_schedule.py

```diff
@@ -364,6 +388,53 @@ def run(self) -> None:
             self.log.fatal("Fatal runtime error: {}\n{}".format(
                 ex, traceback.format_exc()))
 
+    def get_health_checks(self) -> Dict[str, Any]:
+        """
+        Returns a dictionary of health checks based on the synchronization status of images.
+        """
+        unsynced_images = []
+        total_count = 0
+
+        for next_timestamp in self.queue:
+            for image_spec in self.queue[next_timestamp]:
+                next_snapshot = datetime.strptime(next_timestamp, '%Y-%m-%d %H:%M:%S')
+                with self.module.rados.open_ioctx2(int(image_spec.pool_id)) as ioctx:
+                    image_schedule = self.schedules.find(*image_spec)
+                    if not image_schedule:
+                        continue
+                    min_interval = min(image_schedule.items,
+                                       key=lambda x: x[0].minutes)[0].minutes
+                    try:
+                        with rbd.Image(ioctx, image_id=image_spec.image_id,
+                                       read_only=True) as image:
+                            total_count += 1
+                            remote_timestamps = get_remote_timestamp(image)
+                            if remote_timestamps is None:
+                                continue
+                            for timestamp in remote_timestamps:
+                                if timestamp is None:
+                                    continue
+                                time_delta = next_snapshot.timestamp() - timestamp
+                                if time_delta > (min_interval * 60) * 2:
+                                    unsynced_images.append("/".join(map(str, image_spec)))
+                                    break
+                    except rbd.ImageNotFound:
+                        self.log.error(f"image {image_spec.image_id} does not exist")
+                    except Exception as ex:
+                        self.log.error(f"cannot open rbd image {ex}")
+
+        unsynced_count = len(unsynced_images)
+        if total_count > 0 and (unsynced_count / total_count) > 0.1:
+            return {
+                'UNSYNCED_IMAGES': {
```

Comment:
Not sure I like very much the name `UNSYNCED_IMAGES`.  May be just `RBD_MIRROR_SYNC_STATUS`?